### ISSUE TITLE
Convert the Ptolemy II repository from CVS to SVN

### ISSUE THREAD

**Issue by cxbrooks** [cxbrooks/test#71]
Note: the issue was created automatically with bugzilla2github tool

Original bug ID: BZ#139
From: @cxbrooks
Reported version: 7.1.devel
CC:   ian.brown@hsbcib.com, pt-dev@chess.eecs.berkeley.edu

**Comment by cxbrooks** [comment]
We should consider converting from CVS to SVN:
- The NAOMI (http://chess.eecs.berkeley.edu/naomi) is using SVN
in the NAOMI client
- SVN is "more better" than CVS.  That is, SVN was developed
after CVS and has better support for branches etc.
- Ptolemy should include an interface to the version control system.
If we are converting to SVN, then we should do it sooner rather
than later so that the Ptolemy interface would not require both
CVS and SVN.  However, using Eclipse or Netbeans would obviate
the need for this interface.

To convert to SVN we need to:
- Provide explicit instructions on how to use Eclipse with SVN
- Update the nightly build to use SVN
- Convert the CVS repository to SVN and preserve the log files
  (there are scripts to do this)

**Comment by cxbrooks** [comment]
Anon access via http: would be a great advantage of SVN. That would allow me to access the repository via our proxy here and mean that I would not need to continually update and merge manually from snapshots.

**Comment by cxbrooks** [comment]
https access is enabled, see http://chess.eecs.berkeley.edu/ptexternal

The remaining problem is that svn does not recognize the CA for our certificate,
though firefox does.

Another issue is that we need a more complete ChangeLog like what we used
to have.

Another issue is that we need to write scripts that will sanity check
the repository for the proper svn:eol-style and svn:keywords settings.

**Comment by cxbrooks** [comment]
Created attachment 18
Certificate Authority PEM file for source.eecs.berkeley.edu

> Attached file: [sourceBuiltinObjectToken_VerisignClass3PublicPrimaryCertificationAuthority](https://coq.inria.fr/bugfiles/attachment.cgi?id=18) (text/plain, 848 bytes)
> Description:   Certificate Authority PEM file for source.eecs.berkeley.edu

**Comment by cxbrooks** [comment]
About the svn certificate problem, it appears that the issue is 
that the certificate authority (CA) is not in the openssl database.

http://blogs.dovetailsoftware.com/blogs/kmiller/archive/2007/04/12/subversion-and-certificate-authorties.aspx
Discussion about a similar problem, where the solution is for
each user to either accept the certificate permanently, or set the
ssl-authority-files line in ~/.subversion/servers to:
ssl-authority-files=/Users/cxh/tmp/BuiltinObjectToken:VerisignClass3PublicPrimaryCertificationAuthority

where that file is the CA pem file.  To generate the PEM file for the CA:
1) In firefox, go to https://source.eecs.berkeley.edu
2) Click on the lock in the bottom right
3) In the Page Info window, select the Security Tab and then select
View Certificate
4) Select Details and then select the top level CA, which will be something
like VeriSign Class 3 Public Primary CA
5) Export the file as an X.509 PEM file
6) Use that file in ~/.subversion/servers for thev ssl-authority-files value

However, this is just a workaround, and a poor workaround compared to accepting
the certificate permanently.

The real issue is: why doesn't svn (and presumably openssl) know about this CA?

More info on the CA
bash-3.2$ openssl  x509 -text -in
BuiltinObjectToken:VerisignClass3PublicPrimaryCertificationAuthority 
Certificate:
    Data:
        Version: 1 (0x0)
        Serial Number:
            70:ba:e4:1d:10:d9:29:34:b6:38:ca:7b:03:cc:ba:bf
        Signature Algorithm: md2WithRSAEncryption
        Issuer: C=US, O=VeriSign, Inc., OU=Class 3 Public Primary Certification
Authority
        Validity
            Not Before: Jan 29 00:00:00 1996 GMT
            Not After : Aug  1 23:59:59 2028 GMT
        Subject: C=US, O=VeriSign, Inc., OU=Class 3 Public Primary
Certification Authority
        Subject Public Key Info:
            Public Key Algorithm: rsaEncryption
            RSA Public Key: (1024 bit)
                Modulus (1024 bit):
                    00:c9:5c:59:9e:f2:1b:8a:01:14:b4:10:df:04:40:
                    db:e3:57:af:6a:45:40:8f:84:0c:0b:d1:33:d9:d9:
                    11:cf:ee:02:58:1f:25:f7:2a:a8:44:05:aa:ec:03:
                    1f:78:7f:9e:93:b9:9a:00:aa:23:7d:d6:ac:85:a2:
                    63:45:c7:72:27:cc:f4:4c:c6:75:71:d2:39:ef:4f:
                    42:f0:75:df:0a:90:c6:8e:20:6f:98:0f:f8:ac:23:
                    5f:70:29:36:a4:c9:86:e7:b1:9a:20:cb:53:a5:85:
                    e7:3d:be:7d:9a:fe:24:45:33:dc:76:15:ed:0f:a2:
                    71:64:4c:65:2e:81:68:45:a7
                Exponent: 65537 (0x10001)
    Signature Algorithm: md2WithRSAEncryption
        bb:4c:12:2b:cf:2c:26:00:4f:14:13:dd:a6:fb:fc:0a:11:84:
        8c:f3:28:1c:67:92:2f:7c:b6:c5:fa:df:f0:e8:95:bc:1d:8f:
        6c:2c:a8:51:cc:73:d8:a4:c0:53:f0:4e:d6:26:c0:76:01:57:
        81:92:5e:21:f1:d1:b1:ff:e7:d0:21:58:cd:69:17:e3:44:1c:
        9c:19:44:39:89:5c:dc:9c:00:0f:56:8d:02:99:ed:a2:90:45:
        4c:e4:bb:10:a4:3d:f0:32:03:0e:f1:ce:f8:e8:c9:51:8c:e6:
        62:9f:e6:9f:c0:7d:b7:72:9c:c9:36:3a:6b:9f:4e:a8:ff:64:
        0d:64
-----BEGIN CERTIFICATE-----
MIICPDCCAaUCEHC65B0Q2Sk0tjjKewPMur8wDQYJKoZIhvcNAQECBQAwXzELMAkG
A1UEBhMCVVMxFzAVBgNVBAoTDlZlcmlTaWduLCBJbmMuMTcwNQYDVQQLEy5DbGFz
cyAzIFB1YmxpYyBQcmltYXJ5IENlcnRpZmljYXRpb24gQXV0aG9yaXR5MB4XDTk2
MDEyOTAwMDAwMFoXDTI4MDgwMTIzNTk1OVowXzELMAkGA1UEBhMCVVMxFzAVBgNV
BAoTDlZlcmlTaWduLCBJbmMuMTcwNQYDVQQLEy5DbGFzcyAzIFB1YmxpYyBQcmlt
YXJ5IENlcnRpZmljYXRpb24gQXV0aG9yaXR5MIGfMA0GCSqGSIb3DQEBAQUAA4GN
ADCBiQKBgQDJXFme8huKARS0EN8EQNvjV69qRUCPhAwL0TPZ2RHP7gJYHyX3KqhE
BarsAx94f56TuZoAqiN91qyFomNFx3InzPRMxnVx0jnvT0Lwdd8KkMaOIG+YD/is
I19wKTakyYbnsZogy1Olhec9vn2a/iRFM9x2Fe0PonFkTGUugWhFpwIDAQABMA0G
CSqGSIb3DQEBAgUAA4GBALtMEivPLCYATxQT3ab7/AoRhIzzKBxnki98tsX63/Do
lbwdj2wsqFHMc9ikwFPwTtYmwHYBV4GSXiHx0bH/59AhWM1pF+NEHJwZRDmJXNyc
AA9WjQKZ7aKQRUzkuxCkPfAyAw7xzvjoyVGM5mKf5p/AfbdynMk2OmufTqj/ZA1k
-----END CERTIFICATE-----

On the Mac, it looks like OpenSSL keeps certs in
/System/Library/OpenSSL/
One idea would be to edit openssl.cnf and add the CA there.

**Comment by cxbrooks** [comment]
I figured out how to update OpenSSLs certs with the appropriate CA. See

http://chess.eecs.berkeley.edu/ptexternal/wiki/Main/Subversion#svnCertficateIsNotIssuedByATrustedAuthority

This bug remains open pending solving the ChangeLog issue and creating scripts
to sanity check the repository for the proper svn:eol-style and svn:keywords
settings.

**Comment by cxbrooks** [comment]
Closing this:
SVN is installed, see  http://chess.eecs.berkeley.edu/ptexternal/wiki/Main/Subversion
We have https access, see http://chess.eecs.berkeley.edu/ptexternal

I hacked up a way to create ChangeLogs that is fairly fast.  The author
of the svn2cl script suggested operating on files on a per year basis.
I added a rule to $PTII/makefile that creates ChangeLogThisYear.txt
The nightly build will build
http://chess.eecs.berkeley.edu/ptexternal/nightly/ChangeLog.txt
which consists of the ChangeLogs for different years concatenated together.
Only the most recent year is regenerated. 

I created some scripts in adm/bin that find problems with keywords and
update $PTII/doc/coding/releasemgt.htm to describe their use.
